### ISSUE TITLE
Dev Branch: Screen not updating realtime

### ISSUE THREAD

**Issue by rleongcs** [Blackymas/NSPanel_HA_Blueprint#1013]
@edwardtfn I just updated to the latest Dev branch (version 4.0 ready for release) and I am seeing the screen no longer updates real time. For example, if I press the hardware button, the blue line does not turn to gray or vice versa. It only will change if I move to a different screen and come back. The same is for the button pages. If I press any button, the status will not reflect till I move to a different screen and back.

**Comment by edwardtfn** [comment]
Please go to your device's page in Home Assistant (Settings > Devices & Services > ESPHome) and double check if the sensor "Current page" is enabled.

**Comment by rleongcs** [comment]
> Please go to your device's page in Home Assistant (Settings > Devices & Services > ESPHome) and double check if the sensor "Current page" is enabled.

I believed it is not. Do I enable it?

![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/61628312/55006a3e-69e9-4a91-9451-fbe8633aa571)

**Comment by edwardtfn** [comment]
Yes, please enable it.

**Comment by rleongcs** [comment]
Ok...confirmed working after enabling that sensor. Is there any reason it is not automatically enabled?

**Comment by edwardtfn** [comment]
Yeah, you can remove all those entities that are not being provided by ESPHome anymore.

**Comment by edwardtfn** [comment]
Just play as a user will play and let me know if you find some issue. That will help improving the code and the docs. 😉

**Comment by rleongcs** [comment]
> the sensor was already there and it was already disabled (maybe you did that manually)

I never disable any sensor manually. I have just been updating the version. Not sure if it has anything to do with me updating to the version before this latest one and then I saw you pushed this latest update and I again immediately update. Flashed the TFT twice as well. Then there is also the ESPHOME update that was pushed by HA.

**Comment by edwardtfn** [comment]
Yeah... then looks like HA disables the sensor automatically when the integration is not providing it for a while. This means many other users will face the same issue. :(
I will add something to the release notes then.

Thanks a lot!

**Comment by rleongcs** [comment]
> * switch.roomhw_nspanel_relay_1_local_fallback
> * switch.roomhw_nspanel_relay_2_local_fallback

@edwardtfn I just noticed that fallback does not work earlier for the Dev NSPanel when I have to shutdown HA for some time to troubleshoot ip address issues. Pressing the hardware buttons does nothing. However in the Prod NSPanel, the fallback is working. The only difference is the fallback entity is available in the Prod NSPanel. Did the settings change in Dev? And how do I activate it?

Dev
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/61628312/824f89c3-9d3d-4b9d-bd74-d83fd46dda13)

Prod
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/61628312/5679f820-0495-445c-8487-7706808b6f3a)

**Comment by edwardtfn** [comment]
On v4.0 (`dev`) when you assign the entity relative to a relay to a hardware button on the same panel the fallback is automatically enabled.
Isn't that working?

**Comment by rleongcs** [comment]
> Isn't that working?

No...it does not. Below is what I have currently configured.

![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/61628312/665d3bd7-697f-4c37-8ad0-09983443d184)

**Comment by edwardtfn** [comment]
Is "Bath 2" the same as `switch.{{panel_name}}_relay_1"?

**Comment by rleongcs** [comment]
No, it is another switch from a Sonoff DualR3. But I am expecting that when a fallback happens, it should just be turning on or off the relay.

**Comment by edwardtfn** [comment]
Mmmmm
We have rebuilt that engine and this case was not included.

Could you please describe better what is the point of having a different behavior when not connected. I will see how is the better way of implementing a solution (which could be just like it was before).

**Comment by rleongcs** [comment]
> Could you please describe better what is the point of having a different behavior when not connected. I will see how is the better way of implementing a solution (which could be just like it was before).

OMG...looks like the Dev implementation will break all my current Prod setup if I upgrade later on.

My setup is as below...

Room:
NSPanel replaced a 2 gang switch. Each gang has 3 downlights and that makes a total of 6 lights. Out of the 6 lights, 2 are smart downlights while 4 are normal lights that I have connected them to Sonoff Basic to better control the lights using scenes. So now the hardware button is assigned to some scenes. 
During a HA/Network issue, I would still want to be able to on/off the lights that are connected to each of the relays. For the Prod version, it is working as I wanted.

Bath:
NSPanel replaced a 2 gang switch. One gang is controlling both the lights and fan in the bath. Another gang is controlling the hallway lights. In order to control the lights and fan independently, I connected them to a Sonoff DualR3 which is connected to the switch relay. So now a single click on the hardware button turns on/off the lights and a double click turns on/off the fan. The hallway lights is just connected directly for the relay.
During a HA/Network issue, I would still want to be able to on/off the lights and fan that are connected to the relay1. Also on/off the hallway lights.

I realized I am not able to fallback on the hallway lights even though it is the same entity except I have used the "Change device type of a switch".
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/61628312/0d1d01ed-daf1-4946-808d-c19b4a723cf3)
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/61628312/f701d450-0149-4b72-a54e-923fa4c763a3)

**Comment by edwardtfn** [comment]
> OMG...looks like the Dev implementation will break all my current Prod setup if I upgrade later on.

I don't think it would be hard to have it working as before (being able to fallback even when attached to other entity).
I will take a look tomorrow and come back here. 😉

**Comment by edwardtfn** [comment]
How important is for you to have the Fallback settings as switches? Could be just a toggle on the blueprint settings?

A switch is easier if you are using some other automation to set this, but as I believe this shouldn't change dynamically (as you won't be able to change when the panel is offline) I would prefer to have this in the blueprint settings and try to save a bit of memory (our biggest constraints at the moment).

Please let me know your thoughts.

**Comment by rleongcs** [comment]
In the blueprint is fine. Just needed the ability to make sure fallback works when HA/Network is down. Currently, I could not turn on the bathroom lights and imagine the face on my wife having to go to the bathroom in darkness! 😉

**Comment by edwardtfn** [comment]
I've just pushed a fix. 😃 
Unfortunately I cannot test this now, but it would be nice if you can give a try and share your results (with logs if failed, please). 
You have to update your ESPHome and Blueprint. No need to update TFT this time.

**Comment by edwardtfn** [comment]
>  Currently, I could not turn on the bathroom lights and imagine the face on my wife having to go to the bathroom in darkness! 😉

If the panel is online you can still toggle the switch related to the relay you want to enable. :)
It would be a manual thing, for now, but at least you don't have to be in the darkness. 😉

**Comment by rleongcs** [comment]
> I've just pushed a fix. 😃 Unfortunately I cannot test this now, but it would be nice if you can give a try and share your results (with logs if failed, please). You have to update your ESPHome and Blueprint. No need to update TFT this time.

Will test it tomorrow and feedback. 

> If the panel is online you can still toggle the switch related to the relay you want to enable. :)

I totally forgotten about it that I can login to the panel through the url. BTW, what is the login username and password? I totally forgot about it. 😢 
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/61628312/1c32866c-ea70-4be0-b844-67f14135cd7e)

**Comment by edwardtfn** [comment]
```
username: admin
password: ${wifi_password}
```

**Comment by rleongcs** [comment]
> ```
> username: admin
> password: ${wifi_password}
> ```

Thanks...That works! 🎉

**Comment by rleongcs** [comment]
> > I've just pushed a fix. 😃 Unfortunately I cannot test this now, but it would be nice if you can give a try and share your results (with logs if failed, please). You have to update your ESPHome and Blueprint. No need to update TFT this time.
> 
> Will test it tomorrow and feedback.

@edwardtfn Tested and fallback is working perfectly for both the bath relays. Thank you 🎉
